### DISCUSSION TITLE
Can i use Prisma-client as a nested dependency

### DISCUSSION THREAD

**Discussion by Zain-ul-din** [prisma/prisma#17012] [Q&A] [answered]
I was making a restful APIs framework I decided to add ```Prisma``` for database support I want to provide ```Prisma-client``` to users in an appropriate way.
here is what I'm trying to do:

```typescript
// @my/pkg

import {PrismaClient} from '@prisma/client'; // here the prisma client is nested dependency for client

// base class for all routes, the user will extend this class to create a route
abstract class BaseRouter {
    public get (req: Request, res: Response): Promise <void> {}
    
   public  db = new PrimaClient ();
}

 // client side
 // note! prisma init and database will setup on client side
import BaseRouter from '@my/pkg'

class Client extends BaseRouter {
  public get (req: Request, res: Response): Promise <void> {
      const users = db.Users.findMany (); // I'm getting error here
  }
}

```

> so my question is can I use ```pirsma-client``` as a nested dependency and is there any way to achieve similar results?

## Reason for that Design:

I'm providing ```database``` access from the base class because I also want to provide other databases like ```Redis, firebase, etc...``` according to the flavor of the database chosen by the user.
here how it looks like:-
```bash
   which database do you want to choose for your app:
   - Prisma (SQL/MongoDB)
   - firebase (NON-SQL)
   - Redis (caching)
```
According to APIS design, all databases will access through the base class object ```db```.

**Comment by pantharshit00** [comment, marked as answer]
I don't think this will be possible because in order to generate the Prisma client, you will need to know the schema beforehand. Can you tell me if you have access to the models in the package?

**Reply by Zain-ul-din** [reply]
@pantharshit00 thanks for the reply. I decided to adopt the `remix framework` like APIS for Prisma client

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
